src/deeponet.jl

```diff
@@ -0,0 +1,116 @@
+"""
+    DeepONet(; branch = (64, 32, 32, 16), trunk = (1, 8, 8, 16),
+        branch_activation = identity, trunk_activation = identity)
+
+constructs a DeepONet composed of Dense layers. Make sure the last node of `branch` and `trunk` are same.
+
+## Keyword arguments:
+
+- `branch`: Tuple of integers containing the number of nodes in each layer for branch net
+- `trunk`: Tuple of integers containing the number of nodes in each layer for trunk net
+- `branch_activation`: activation function for branch net
+- `trunk_activation`: activation function for trunk net
+
+## References:
+Lu Lu, Pengzhan Jin, George Em Karniadakis , "DeepONet: Learning nonlinear operators for identifying differential equations based on the universal approximation theorem of operators"
+doi: https://arxiv.org/abs/1910.03193
+
+## Example:
+```julia-repl
+julia> deeponet = DeepONet(; branch = (64, 32, 32, 16), trunk = (1, 8, 8, 16))
+
+Branch : Chain(
+    layer_1 = Dense(64 => 32),          # 2_080 parameters
+    layer_2 = Dense(32 => 32),          # 1_056 parameters
+    layer_3 = Dense(32 => 16),          # 528 parameters
+)         # Total: 3_664 parameters,
+          #        plus 0 states.
+ 
+  Trunk: Chain(
+    layer_1 = Dense(1 => 8),            # 16 parameters
+    layer_2 = Dense(8 => 8),            # 72 parameters
+    layer_3 = Dense(8 => 16),           # 144 parameters
+)         # Total: 232 parameters,
+          #        plus 0 states.
+()  # 3_896 parameters
+```
+"""
+function DeepONet(; branch = (64, 32, 32, 16), trunk = (1, 8, 8, 16),
+    branch_activation = identity, trunk_activation = identity)
+
+    # checks for last dimension size
+    @assert branch[end] == trunk[end] "Branch and Trunk net must share the same amount of nodes in the last layer. 
+    Otherwise Σᵢ bᵢⱼ tᵢₖ won't work."
+
+    branch_net = Chain([Dense(branch[i] => branch[i+1], branch_activation)
+    for i in 1:length(branch) - 1]...);
```

Comment:
You dont need semi colons

Reply:
Sure! Curious if there are any specific reasons for not doing so, like what would the difference be?

Reply:
for uniform style, in julia you rarely use semicolons. the only place with a justified usecase for that is if you really want to cram multiple statements in a single line (which might also be questionable in the first place)